### PR TITLE
Added support for dynamic port

### LINKED ISSUE #1571: Make desktop hot reload use a different port than web
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/master/CONTRIBUTING.md) for onboarding!
___

**Problem**
We recently introduced a hotreload feature for the desktop platform. It works with webpack, and is hard-coded to listen on localhost:8080.

In most cases that works fine, but let's say you have some feature that's somehow different on web vs desktop, and you are trying to test that feature by running web and desktop at the same time. If you first run npm run web, it will do a web-specific webpack build, and host that with the webpack dev server on localhost:8080. Then, you run npm run desktop, and it will do a desktop-specific webpack build, and host that with the webpack dev server on localhost:8081. I think webpack does this automatically – if you specify port 8080 but it's already occupied, it will just use port 8081. But the electron client will be connected to localhost:8080 regardless, and thus will be using the web-specific code.

**Why is this important?**
This happened to @marcaaron while he was testing something, and it's confusing to figure out without the context of how the desktop hot reload feature works.

**Solution**
Make desktop's hot reload feature listen on a different port dynamically if the webpack build ends up using a different port that 8080.

### PR DESCRIPTION
**Pull request by parasharrajat**
Please review @roryabraham @marcaaron 

### Details
Added a script under the desktop folder named `start.js` which runs the same command as the npm script to build the desktop dev server but we first find an available port which is then used to run the Renderer process.

### Fixed Issues
<Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing>
Fixes #1571 

### Tests
 1. Run the webserver using `npm run web`. Wait for it to successfully start and then note the Port used.
 2. Run the Desktop server using `npm run desktop` which should use a different port than the port used on the web process.
 3. To test the regressions, run various servers to block the Ports and then run the desktop server. It should start with an available port.

### Tested On

- [ ] Web
- [ ] Mobile Web
- [x] Desktop
- [ ] iOS
- [ ] Android

### Screenshots
N/A

### CODE REVIEW

File: desktop/start.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env node
```
Comment (roryabraham) [inline desktop/start.js:1]: heh, I didn't know you could write node scripts with a shebang like this.

**Review by roryabraham — COMMENTED** [review]
Code looks good, but unfortunately it looks like the desktop-specific code isn't working (i.e: `--platform desktop` isn't being used in the webpack build), unless it's running alone.

I _think_ this is happening because the same webpack build is being reused for different ports, when we really need separate webpack builds for web vs desktop.

File: desktop/start.js (hunk 1 above)
Comment (parasharrajat) [reply desktop/start.js:1]: Ah. It only works for Unix. But we really don't need this line so I will remove it.

**Comment by parasharrajat** [comment]
I forgot to pass the port to the main process it was using the same for the web which is by default set to 8080. Updating.

**Comment by parasharrajat** [comment]
@roryabraham  Updated. Must be working now.

**Review by roryabraham — APPROVED** [review]
LGTM, thanks for the quick turnaround 👍

**Comment by laurenreidexpensify** [comment]
Nice one @parasharrajat - we'll issue payment for this in 7days after no regressions.

**Pull request merged**